### ISSUE TITLE
Chinese-LLaMA-Alpaca-2预训练微调OOM

### ISSUE THREAD

**Issue by Bellocccc** [ymcui/Chinese-LLaMA-Alpaca-2#257]
### 提交前必须检查以下项目

- [X] 请确保使用的是仓库最新代码（git pull），一些问题已被解决和修复。
- [X] 我已阅读[项目文档](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki)和[FAQ章节](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki/常见问题)并且已在Issue中对问题进行了搜索，没有找到相似问题和解决方案。
- [X] 第三方插件问题：例如[llama.cpp](https://github.com/ggerganov/llama.cpp)、[LangChain](https://github.com/hwchase17/langchain)、[text-generation-webui](https://github.com/oobabooga/text-generation-webui)等，同时建议到对应的项目中查找解决方案。

### 问题类型

模型训练与精调

### 基础模型

Chinese-Alpaca-2 (7B/13B)

### 操作系统

Linux

### 详细描述问题

```
# 请在此处粘贴运行代码（请粘贴在本代码块里）
```
# 运行脚本前请仔细阅读wiki(https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki/pt_scripts_zh)
# Read the wiki(https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki/pt_scripts_zh) carefully before running the script
lr=2e-4
lora_rank=64
lora_alpha=128
lora_trainable="q_proj,v_proj,k_proj,o_proj,gate_proj,down_proj,up_proj"
#modules_to_save="embed_tokens,lm_head"
lora_dropout=0.05

pretrained_model=/usr/Chinese-LLaMA-Alpaca-2-main/Chinese-LLaMA-Alpaca-2-main/scripts/langchain/chinese-alpaca-2-7b
chinese_tokenizer_path=/usr/Chinese-LLaMA-Alpaca-2-main/Chinese-LLaMA-Alpaca-2-main/scripts/langchain/chinese-alpaca-2-7b
dataset_dir=/usr/Chinese-LLaMA-Alpaca-2-main/Chinese-LLaMA-Alpaca-2-main/scripts/langchain/tran_data
data_cache=temp_data_cache_dir
per_device_train_batch_size=1
gradient_accumulation_steps=4
block_size=128
output_dir=/usr/Chinese-LLaMA-Alpaca-2-main/Chinese-LLaMA-Alpaca-2-main/scripts/langchain/output

deepspeed_config_file=ds_zero2_no_offload.json

torchrun --nnodes 1 --nproc_per_node 1 run_clm_pt_with_peft.py \
    --deepspeed ${deepspeed_config_file} \
    --model_name_or_path ${pretrained_model} \
    --tokenizer_name_or_path ${chinese_tokenizer_path} \
    --dataset_dir ${dataset_dir} \
    --data_cache_dir ${data_cache} \
    --validation_split_percentage 0.001 \
    --per_device_train_batch_size ${per_device_train_batch_size} \
    --do_train \
    --seed $RANDOM \
    --fp16 \
    --num_train_epochs 1 \
    --lr_scheduler_type cosine \
    --learning_rate ${lr} \
    --warmup_ratio 0.05 \
    --weight_decay 0.01 \
    --logging_strategy steps \
    --logging_steps 10 \
    --save_strategy steps \
    --save_total_limit 3 \
    --save_steps 100 \
    --gradient_accumulation_steps ${gradient_accumulation_steps} \
    --preprocessing_num_workers 8 \
    --block_size ${block_size} \
    --output_dir ${output_dir} \
    --overwrite_output_dir \
    --ddp_timeout 30000 \
    --logging_first_step True \
    --lora_rank ${lora_rank} \
    --lora_alpha ${lora_alpha} \
    --trainable ${lora_trainable} \
    --lora_dropout ${lora_dropout} \
    #--modules_to_save ${modules_to_save} \
    --torch_dtype float16 \
    --load_in_kbits 16 \
    --gradient_checkpointing True \
    --ddp_find_unused_parameters False

### 依赖情况（代码类问题务必提供）

```
# 请在此处粘贴依赖情况（请粘贴在本代码块里）
```
bitsandbytes             0.41.0
peft                     0.3.0.dev0
sentence-transformers    2.2.2
sentencepiece            0.1.97
torch                    2.0.1
torchvision              0.15.2
transformers             4.31.0

### 运行日志或截图

```
# 请在此处粘贴运行日志（请粘贴在本代码块里）
```
![image](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/assets/144331966/94897fd5-a646-451f-9af3-3a8a7ba97309)

**Comment by yufanghui** [comment]
+1

**Comment by iMountTai** [comment]
alpaca-2-7b模型权重只有13g左右，加载模型就OOM，GPU上面有另外的程序运行吧？

**Comment by yufanghui** [comment]
> alpaca-2-7b模型权重只有13g左右，加载模型就OOM，GPU上面有另外的程序运行吧？

破案了，是因为我用的13b的原因，24g显存，2卡的遭不住，换成7b就可以了
